cdc/log.cc

```diff
@@ -900,18 +1044,26 @@ class transformer final {
 
         // TODO: this assumes all mutations touch the same set of columns. This might not be true, and we may need to do more horrible set operation here.
         if (!p.static_row().empty()) {
-            p.static_row().get().for_each_cell([&] (column_id id, const atomic_cell_or_collection&) {
-                auto& cdef =_schema->column_at(column_kind::static_column, id);
-                static_columns.emplace_back(id);
-                columns.emplace_back(&cdef);
-            });
+            // for postimage we need everything...
+            if (_schema->cdc_options().postimage()) {
+                for (const column_definition& c: _schema->static_columns()) {
+                    static_columns.emplace_back(c.id);
+                    columns.emplace_back(&c);
+                }
+            } else {
+                p.static_row().get().for_each_cell([&] (column_id id, const atomic_cell_or_collection&) {
+                    auto& cdef =_schema->column_at(column_kind::static_column, id);
+                    static_columns.emplace_back(id);
+                    columns.emplace_back(&cdef);
+                });
+            }
         }
         if (!p.clustered_rows().empty()) {
             const bool has_row_delete = std::any_of(p.clustered_rows().begin(), p.clustered_rows().end(), [] (const rows_entry& re) {
                 return re.row().deleted_at();
             });
-
-            if (has_row_delete) {
+            // for postimage we need everything...
+            if (has_row_delete || _schema->cdc_options().postimage()) {
```

Comment:
Hmm, maybe we should just make it consistent and always fetch the entire row for preimage?
It would be good conceptually (probably), but with collections it could reduce performance significantly... so I'm not sure.

Reply:
In my opinion, it's better for preimage to work the same way as postimage and always show full row.

Reply:
I think pre-image should be minimal. It basically means the user says he will trade some less storage overhead for doing the merge/interpret merge into postimage himself.

Reply:
I guess having one of preimage/postimage being full and the other being partial is a reasonable midpoint. If user wants whole record for a change then they use postimage if they are interested only in changed bits then preimage + delta deliver just that. So I guess, I agree with @elcallio here after all :)